### PR TITLE
terraform: fix checksum

### PR DESCRIPTION
**Pull request by b4nst**
#### Description

It seems terraform had an outage regarding checksums and changed them. It breaks the installation. I started to fix some but we still need to update all of them.

Since such a change is not expected, we must also double check those checksums are not fishy

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.2 22D49 arm64
Command Line Tools 14.2.0.0.1.1668646533

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by b4nst** [comment]
[Similar incident for linux packages](https://status.hashicorp.com/incidents/khlnhcbyvzsr)

File: sysutils/terraform/Portfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (herbygillot) [inline sysutils/terraform/Portfile:1]: The modeline (`# -*- ...`) should be the first line in the `Portfile`

File: sysutils/terraform/Portfile [hunk 2]
```diff
@@ -44,9 +45,9 @@ subport terraform-1.3 {
                         sha256  aa111cd80d84586697d1643c6c21452d34f70e5bc639e4106856f59382351397 \
                         size    20852937 \
                         [terraformDistBase]_arm64.zip \
```
Comment (herbygillot) [inline sysutils/terraform/Portfile:47]: Have you confirmed that the `amd64` sums don't need to be changed either?

**Comment by harens** [comment]
Should this not be treated similar to a [stealth update](https://trac.macports.org/wiki/PortfileRecipes#stealth-updates)?

**Comment by AndyMoore** [comment]
same for older versions of terraform, eg

```--->  Verifying checksums for terraform-1.2
Error: Checksum (rmd160) mismatch for terraform_1.2.9_darwin_arm64.zip
Error: Checksum (sha256) mismatch for terraform_1.2.9_darwin_arm64.zip
Error: Checksum (size) mismatch for terraform_1.2.9_darwin_arm64.zip
Error: Failed to checksum terraform-1.2: Unable to verify file checksums
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_terraform/terraform-1.2/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
Error: Processing of port terraform-1.2 failed
```

although this is marked as obsolete, it's causing problems elsewhere (eg latest terragrunt uses terraform-1.2 as a runtime dependency)

**Pull request merged**